backend/pkg/server/api/handler.go

```diff
@@ -108,76 +111,95 @@ func (h *Handler) GetResourceDetails(ctx *gin.Context) {
 func (h *Handler) GetCustomBoard(ctx *gin.Context) {
 	var err error
 
-	config, ok := h.boards[ctx.Param("id")]
-	if !ok {
-		ctx.AbortWithStatus(http.StatusNotFound)
-		return
-	}
+	session := sessions.Default(ctx)
 
-	endpoints, ok := h.clients[ctx.Param("cluster")]
-	if !ok {
-		ctx.AbortWithStatus(http.StatusNotFound)
-		return
-	}
+    // Get the profile from the session
+    profile := session.Get("profile")
 
-	query := ctx.Request.URL.Query()
+	userProfile, ok := profile.(auth.Profile)
+    if !ok {
+        ctx.JSON(500, gin.H{"error": "invalid profile data"})
+        return
+    }
 
-	sources := config.Sources.List
-	if len(sources) > 0 {
-		query["sources"] = sources
-	}
+    // Now you can access individual fields of the profile
+    email := userProfile.Email
+	config, ok := h.boards[ctx.Param("id")]
+	users := config.Users.List
 
-	if len(sources) == 0 {
-		sources, err = endpoints.Core.ListSources(ctx, url.Values{})
-		if err != nil {
-			ctx.AbortWithError(http.StatusInternalServerError, err)
+    if slices.Contains(users, email) {
```

Comment:
what happens when a custom board has an empty user list? In this case it should also be allowed.

Another suggestion would be to use a guard clause to make the code easier.

```suggestion
    if !slices.Contains(users, email) {
        ctx.AbortWithStatus(http.StatusUnauthorized)
        return
    }
```